app/controllers/status_updates_controller.rb

```diff
@@ -23,6 +23,13 @@ def create
       UnsubscribeAllService.call(subscriber, :non_existent_email) if subscriber
     end
 
+    begin
+      Email.where(id: reference).update!(notify_status: status)
```

Comment:
`Email.where(id: reference).update!(notify_status: status)` ->  `Email.find(reference).update!(notify_status: status)`

Reply:
Yeah... I have one very mild reservation about this, which is that find returns an array and I can't call explain on it to check how it's doing the find. But I'm going to assume it does the right thing behind the scenes, so I'll make the change.

Reply:
I think `find` will only return an array if an array is passed to it, passing just a string should just return the email object (if it exists), but yeah you can't find the query information from the return object. However, I'd be very surprised if `find` didn't use the newly created index (assume that's what you're verifying). Could also check this via Postgres logs.